lib/serve/functions.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+var _ = require('lodash');
+var chalk = require('chalk');
+var path = require('path');
+var RSVP = require('rsvp');
+var getProjectId = require('../getProjectId');
+var utils = require('../utils');
+var parseTriggers = require('../parseTriggers');
+var EmulatorController;
+var controller;
+
+function _pollOperation(op) {
```

Comment:
Is there any of the logic between this and the other pollOperation logic in deploy that can be abstracted / shared? Not a big deal but feels a little off to have to write all this logic twice.